hal-x86_64/src/framebuffer.rs

```diff
@@ -0,0 +1,101 @@
+use core::ptr;
+use hal_core::framebuffer::{Draw, RgbColor};
+
+#[derive(Debug)]
+pub struct Framebuffer<'buf, B> {
+    buf: B,
+    cfg: &'buf Config,
+}
+
+#[derive(Debug, Clone)]
+pub struct Config {
+    /// The framebuffer height in pixels
+    pub height: usize,
+    /// The framebuffer width in pixels.
+    pub width: usize,
+    pub px_bytes: usize,
+    pub line_len: usize,
+    pub px_kind: PixelKind,
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum PixelKind {
+    Bgr,
+    Rgb,
+    Gray,
+}
+
+#[cfg(feature = "embedded-graphics-core")]
+#[doc(cfg(feature = "embedded-graphics-core"))]
+mod embedded_graphics;
+#[cfg(feature = "embedded-graphics-core")]
+#[doc(cfg(feature = "embedded-graphics-core"))]
+pub use self::embedded_graphics::*;
```

Comment:
Perhaps this should be dropped, as I don't think this is where it's going to end up. I suppose it's deleted in #114 so it doesn't matter.